packages/vite-plugin-svelte/src/utils/compile.js

```diff
@@ -133,6 +133,31 @@ export function createCompileSvelte() {
 		let compiled;
 		try {
 			compiled = svelte.compile(finalCode, { ...finalCompileOptions, filename });
+			if (compiled.css) {
+				if (finalCode.includes(':global') && compiled.ast.css) {
```

Comment:
Could we have a comment here of what this is trying to do? Is it checking for query selectors that starts with `:global()` and not the nested kind?

Also a bit concerned of the perf implication with the additional walking but I guess it can be optimized later.

Reply:
yes, unscoped `:global` must be included or it would be a breaking change and also differ in experience between dev and build because in dev vite still includes the css, its only treeshaken in build.

the include check before walking should minimize impact. If you feel this is still too much of a regression, should we make this opt-in?